### PR TITLE
Completed Some Progress Routes

### LINKED ISSUE #19: Create Backend Route to Get Participants Progress
- Progress Route
- Protected Procedure

### LINKED ISSUE #20: Create Backend Route to Get Participants Progress for a Room
- Progress Route
- Protected Procedure

### PR DESCRIPTION
**Pull request by HARI-PRMD**
Closes #19, Closes #20

### CODE REVIEW

File: src/server/api/routers/progress.ts [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+            select: {
+              id: true,
+              name: true,
+              image: true,
+            },
+          },
+        },
+      });
+
+      if (!room) {
```
Comment (Three6ty1) [inline src/server/api/routers/progress.ts]: nah imm ado my own thing

**Review by Three6ty1 — APPROVED** [review]
lgtm 
image

File: src/server/api/routers/progress.ts (hunk 1 above)
Comment (HARI-PRMD) [reply src/server/api/routers/progress.ts]: gotcha one sec 👍 
Screenshot 2024-02-01 at 1 25 42 am

File: src/server/api/routers/progress.ts [hunk 2]
```diff
@@ -76,12 +75,6 @@ export const ProgressRouter = createTRPCRouter({
         },
       });
 
-      if (!room) {
```
Comment (Three6ty1) [inline src/server/api/routers/progress.ts]: lgtm

**Review by Three6ty1 — APPROVED** [review]
image

File: src/server/api/routers/progress.ts (hunk 2 above)
Comment (HARI-PRMD) [reply src/server/api/routers/progress.ts]: Screenshot 2024-02-01 at 1 28 08 am

**Pull request merged**